app/core/api_client.py

```diff
@@ -41,78 +49,112 @@ async def _make_request(
         """
         try:
             response = await self.client.request(
-                method=method,
-                headers=headers,
-                url=endpoint,
-                params=params,
-                json=data
+                method=method, headers=headers, url=endpoint, params=params, json=data
             )
             response.raise_for_status()
             # Check if the response is JSON by inspecting the Content-Type header
-            if response.headers.get("Content-Type") == "application/json":
+            if "application/json" in response.headers.get("Content-Type"):
```

Comment:
So wdyt about  `response.headers.get("Content-Type", "").startswith("application/json")` ?

Reply:
`response.headers.get("Content-Type", "").startswith("application/json")`
 is probably more precise because it checks specifically that the `Content-Type` starts with “application/json”. That way, we are double sure that “application/json” appears only as the initial part of the “Content-Type”



`if "application/json" in response.headers.get("Content-Type", “”):`

is, on the other hand, more flexible. It matches "application/json" even if it appears in a different part of the "Content-Type."   
Considering the current case, even if both solutions are valid, I think yours is more clear.